### DISCUSSION TITLE
Performance Bottlnecks

### DISCUSSION THREAD

**Discussion by gorentbarak** [spacedriveapp/spacedrive#1414] [Community Support] [answered]
What are the main performance bottlenecks here? I'm interesting in contributing and would like to see where I could help.

**Comment by fogodev** [comment]
Which kind of performance bootlenecks you're looking for? Front-end or Back-end to begin with?

**Reply by gorentbarak** [reply]
I'm better at Rust, so back-end.

**Reply by fogodev** [reply, marked as answer]
Have to think a bit then lol, I'm usually the one doing performance improvs in our Rust core directly (not counting improvements on our libs used by core).  Our most intensive loads are on some jobs, like the indexer job which needs to walk through possibly massive directories and subdirectories. Also, the media processor job have a bunch of CPU-bound work to extract media data and generate thumbnails. The later one possibly has some low-hanging fruits to be taken, like some missing `tokio::spawn_blocking` to process blocking CPU stuff outside from tokio's futures executor. 

The indexer is a bit harder to improve, I had to introduce a bunch of strategies to achieve an OK scalability with it. Like processing new-found subdirectories on new steps of our job system. But new ideas are always welcome.
